### DISCUSSION TITLE
Enabled property in useSimulateHook in wagmi 2

### DISCUSSION THREAD

**Discussion by wahabshah21** [wevm/wagmi#4340] [Q&A] [answered]
I guess wagmi 2 doesnt have enabled property for the useSimulateHook. WHat other way is there to stop the hook from running the first time on component load? Currently it automatically gets called on load, which I dont want. Is it posisible somehow or any alternative to enabled property?

**Comment by jxom** [comment]
Have you tried the `query.enabled` property?

**Reply by wahabshah21** [reply]
No I haven't, is it passed inside the object passed to useSimulateHook? Can you point me in the docs or just where is this added?

Sharing an example form the docs below. Does this query.enabled go somewhere in there? Any hints would be appreciated

```
function App() {
  const result = useSimulateContract({
    abi,
    address: '0x6b175474e89094c44da98b954eedeac495271d0f',
    functionName: 'transferFrom',
    args: [
      '0xd2135CfB216b74109775236E36d4b433F1DF507B',
      '0xA0Cf798816D4b9b9866b5330EEa46a18382f251e',
      123n,
    ],
  })
}
```

**Reply by tmm** [reply, marked as answer]
```ts
function App() {
  const [enabled, setEnabled] = React.useState(false)
  const result = useSimulateContract({
    abi,
    address: '0x6b175474e89094c44da98b954eedeac495271d0f',
    functionName: 'transferFrom',
    args: [
      '0xd2135CfB216b74109775236E36d4b433F1DF507B',
      '0xA0Cf798816D4b9b9866b5330EEa46a18382f251e',
      123n,
    ],
    query: {
      enabled,
    },
  })
}
```

**Reply by wahabshah21** [reply]
Yes this works, thanks for sharing  @jxom @tmm
